firmware/hw_layer/sensors/hip9011_logic.cpp

```diff
@@ -183,3 +187,28 @@ void HIP9011::handleSettings(int rpm DEFINE_PARAM_SUFFIX(DEFINE_HIP_PARAMS)) {
 			prescaler = new_prescaler;
 	}
 }
+
+int HIP9011::cylinderToChannelIdx(int cylinder) {
+	/* TODO: hip9011 inputs to bank mapping? */
+	return getCylinderKnockBank(cylinder);
+}
+
+void HIP9011::handleChannel(DEFINE_HIP_PARAMS) {
+	int ret;
+
+	/* we did not receive any callback from spark logic with valid cylinder yet */
+	if (cylinderNumber < 0)
+		return;
+
+	/* find next firing cylinder */
+	/* MAGIC +1 -1, couse getNextFiringCylinderId expect cylinders to start from 1 */
+	expectedCylinderNumber = getNextFiringCylinderId((cylinderNumber + 1) PASS_ENGINE_PARAMETER_SUFFIX) - 1;
```

Comment:
why do we need next expected cylinder?

Reply:
To setup hip9011 for next cylinder as soon as possible. And when times come - we only set gpio starting integration.

Reply:
ah, so we don't have to make an SPI transaction to select the correct channel at the instant we start sampling, got it